### DISCUSSION TITLE
Use other voxel grid types?

### DISCUSSION THREAD

**Discussion by SpatialDigger** [pyvista/pyvista#3758] [Q&A]
As far as I understand it Voxeliser defaults to a cube grid.
I use `data = PVGeo.filters.VoxelizePoints().apply(points)`
This gives a regular cube grid (note: if all points are equidistant)

Are there any options to use different 3D grids, e.g. 3D hex grids? Tetrahedral grids? If possible is this best achieved through PVGeo or pyVista?

Thanks

General ponderings follow:

The following link demonstrates the conversion of a rectilinear grid to tetrahedra, I have an unstructured mesh, can this be 'voxelised' to a tetra-grid?
https://docs.pyvista.org/api/core/_autosummary/pyvista.RectilinearGrid.to_tetrahedra.html#to-tetrahedra

**Comment by banesullivan** [comment]
This depends... it can be a tough problem to voxelize or create other 3D cell types from a point cloud. Perhaps try https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSetFilters.delaunay_3d.html#delaunay-3d
